utility/resource_loader_compat.h

```diff
@@ -0,0 +1,281 @@
+		_err_print_error(FUNCTION_STR, __FILE__, __LINE__, "Condition \"" _STR(m_cond) "\" is true. Returning: " _STR(m_retval), DEBUG_STR(m_msg)); \
+		return m_retval;                                                                                                                            \
+	} else                                                                                                                                          \
+		((void)0)
+
+#define ERR_RFLBC_COND_V_CLEANUP(m_cond, m_retval, loader)                                                                                              \
+	if (unlikely(m_cond)) {                                                                                                                         \
+	    if (loader != nullptr) memdelete(loader);                                                                                                   \
+		return m_retval;                                                                                                                            \
+	} else                                                                                                                                          \
+		((void)0)
+
+
+struct ResourceProperty{
+	String name;
+	Variant::Type type;
+	Variant value;
+};
+
+class ResourceLoaderBinaryCompat {
+    bool translation_remapped = false;
+	bool fake_load = false;
+	// The localized (i.e. res://) path to the resource
+	String local_path;
+	// The actual path to the resource (either "res://" in pack or file system)
+	String res_path;
+
+	String type;
+	String project_dir;
+	Ref<Resource> resource;
+	Ref<ResourceImportMetadatav2> imd;
+	uint32_t ver_format = 0;
+	uint32_t engine_ver_major = 0;
+	uint32_t engine_ver_minor = 0;
+	bool stored_big_endian = false;
+	bool stored_use_real64 = false;
+	bool convert_v2image_indexed = false;
+	bool hacks_for_deprecated_v2img_formats = true;
+	bool no_abort_on_ext_load_fail = true;
+	FileAccess *f = nullptr;
+
+	uint64_t importmd_ofs = 0;
+
+	Vector<char> str_buf;
+	Vector<StringName> string_map;
+
+	StringName _get_string();
+
+	struct ExtResource {
+		String path;
+		String type;
+		RES cache;
+	};
+
+	bool use_sub_threads = false;
+	float *progress = nullptr;
+	Vector<ExtResource> external_resources;
+	struct PropPair {
+		StringName name;
+		Variant value;
+	};
+	struct IntResource {
+		String path;
+		uint64_t offset;
+	};
+
+
+	Vector<IntResource> internal_resources;
+	Map<String, RES> internal_index_cache;
+	Map<String, String> internal_type_cache;
+	Map<String, List<ResourceProperty>> internal_index_cached_properties;
+
+	ResourceFormatLoader::CacheMode cache_mode = ResourceFormatLoader::CACHE_MODE_IGNORE;
+	void save_unicode_string(const String &p_string);
+	static void save_ustring(FileAccess * f, const String &p_string);
+	Error repair_property(const String &rtype, StringName &name, Variant &value);
```

Comment:
```suggestion
	Error repair_property(const String &rtype, const StringName &name, Variant &value);
```

Same as `ensure_dir`, it will cause multiple `error: non-const lvalue reference to type 'StringName' cannot bind to a temporary of type 'StringName'` build errors.

Reply:
damn, I totally should have tested this on Linux with gcc, MSVC never gave me errors about these inconsistent parameter definitions😔